File: android_mobile/src/main/java/com/alexstyl/specialdates/upcoming/ContactViewModelFactory.java
package com.alexstyl.specialdates.upcoming;

import android.graphics.Typeface;
import android.view.View;

import com.alexstyl.resources.ColorResources;
import com.alexstyl.resources.StringResources;
import com.alexstyl.specialdates.contact.Contact;
import com.alexstyl.specialdates.date.ContactEvent;
import com.alexstyl.specialdates.date.Date;

public final class ContactViewModelFactory {
    private final ColorResources colorResources;
    private final StringResources stringResources;

    public ContactViewModelFactory(ColorResources colorResources, StringResources stringResources) {
        this.colorResources = colorResources;
        this.stringResources = stringResources;
    }

    ContactEventViewModel createViewModelFor(Typeface typeface, Date date, ContactEvent contactEvent) {
        Contact contact = contactEvent.getContact();
        return new ContactEventViewModel(
                contact,
                View.VISIBLE,
                contact.getDisplayName().toString(),
                contactEvent.getLabel(date, stringResources),
                colorResources.getColor(contactEvent.getType().getColorRes()),
                (int) contact.getContactID(),
                contact.getImagePath(),
                typeface
        );
    }

}

File: android_mobile/src/main/java/com/alexstyl/specialdates/upcoming/ContactViewModelFactory.kt
package com.alexstyl.specialdates.upcoming

import com.alexstyl.resources.ColorResources
import com.alexstyl.specialdates.Strings
import com.alexstyl.specialdates.date.ContactEvent
import com.alexstyl.specialdates.date.Date

class ContactViewModelFactory(private val colorResources: ColorResources, private val strings: Strings) {

    internal fun createViewModelFor(date: Date, contactEvent: ContactEvent): UpcomingContactEventViewModel {
        val contact = contactEvent.contact
        return UpcomingContactEventViewModel(
                contact,
                contact.displayName.toString(),
                contactEvent.getLabel(date, strings),
                colorResources.getColor(EventColors.colorOf(contactEvent.type)),
                contact.contactID.toInt(),
                contact.imagePath
        )
    }


}
